File: backend/user-service/src/main/java/io/kgu/userservice/domain/response/ResponseUser.java
package io.kgu.userservice.domain.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class ResponseUser {

    private String userId;
    private String email;
    private String name;
    private String statusMessage;
    private String picture;
    private List<ResponseUser> userFriends;
    private List<ResponseUser> userBlocked;

}

File: android/RandHandChat/app/src/main/java/com/kyonggi/randhand_chat/Domain/User/ResponseUser.kt
package com.kyonggi.randhand_chat.Domain.User

import com.google.gson.annotations.SerializedName

data class ResponseUser(
    @SerializedName("userId")
    var userId: String?,
    @SerializedName("email")
    var email: String?,
    @SerializedName("name")
    var name: String?,
    @SerializedName("statusMessage")
    var message: String?,
    @SerializedName("picture")
    var picture: String?,
    @SerializedName("userFriends")
    var userfriends: List<ResponseUser>?,
    @SerializedName("userBlocked")
    var userblocked: List<ResponseUser>?
)